tests/e2e/hci.go

```diff
@@ -488,4 +495,335 @@ var _ bool = ginkgo.Describe("hci", func() {
+		var datacenters []string
+		cfg, err := getConfig()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		dcList := strings.Split(cfg.Global.Datacenters,
+			",")
+		for _, dc := range dcList {
+			dcName := strings.TrimSpace(dc)
+			if dcName != "" {
+				datacenters = append(datacenters, dcName)
+			}
+		}
+
+		for _, dc := range datacenters {
+			defDc, err := finder.Datacenter(ctx, dc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			finder.SetDatacenter(defDc)
+			hosts, err := finder.HostSystemList(ctx, "*")
+			for _, host := range hosts {
+				if host.Reference().Reference().Value == targetHost.Value {
+					hostInfo := host.Common.InventoryPath
+					hostIpInfo := strings.Split(hostInfo, "/")
+					hostIp = hostIpInfo[len(hostIpInfo)-1]
+					break
+				}
+			}
+		}
+
+		framework.Logf("hostIp: %s", hostIp)
+
+		toggleNetworkFailureParallel([]string{hostIp}, true)
+		isTargetHostIsolated = true
+
+		defer func() {
+			if isTargetHostIsolated {
+				ginkgo.By("Re-integrate isolated host back into vsan cluster before terminating test")
+				toggleNetworkFailureParallel([]string{hostIp}, false)
+				isTargetHostIsolated = false
+			}
+		}()
+
+		ginkgo.By("Wait for 5-10 mins, verify that the k8s-worker is restarted and brought up on another host")
+		wait4AllK8sNodesToBeUp(nodes)
+		gomega.Expect(waitForAllNodes2BeReady(ctx, client)).To(gomega.Succeed())
+
+		time.Sleep(5 * time.Minute)
+
+		ginkgo.By("Scale up statefulset and scale down statefulset1 to 3 replicas")
+		scaleUpStsAndVerifyPodMetadata(ctx, client, namespace, sts, replicas+2, true, true)
+		gomega.Expect(fss.CheckMount(ctx, client, sts, mountPath)).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Re-integrate isolated host back into vsan cluster")
+		if isTargetHostIsolated {
+			toggleNetworkFailureParallel([]string{hostIp}, false)
+			isTargetHostIsolated = false
+		}
+
+		ginkgo.By("Scale up statefulset to 5 replicas")
+		scaleUpStsAndVerifyPodMetadata(ctx, client, namespace, sts, replicas+2, true, true)
+		gomega.Expect(fss.CheckMount(ctx, client, sts, mountPath)).NotTo(gomega.HaveOccurred())
+
+	})
+
+	/*
+		k8s worker node power off
+		steps:
+		1.	Create an environment as described in the testbed layout above
+		2.  Create statefulset with 1 replica.
+		3.  Isolate a host which has a k8s-worker with attached PVs in cluster1.
+		4.  Wait for 5 mins, verify that the k8s-worker is restarted and brought up on another host.
+		5.  Verify that the replicas on the k8s-worker come up successfully.
+		6.  Scale up statefulset to 3 replicas.
+		7.  Re-integrate the host used in step 3
+		8.  Scale up statefulset to 5 replicas and verify they are successful
+		9.  Cleanup all objects created during the test
+	*/
+	ginkgo.It("Power off k8s worker with attached PVs", ginkgo.Label(p0, block, vanilla, hci), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var err error
+		var replicas int32 = 3
```

Comment:
Description of the test says 1 replica - and here it is initialised to 3

Reply:
Updated